guides/docs/testing/testing_controllers.md

```diff
@@ -133,169 +222,489 @@ defmodule HelloWeb.Router do
 
   pipeline :api do
     plug :accepts, ["json"]
-    resources "/users", HelloWeb.UserController
   end
 
-  ...
+  scope "/", HelloWeb do
+    pipe_through :browser # Use the default browser stack
+
+    get "/", PageController, :index
+  end
+
+  # Other scopes may use custom stacks.
+  scope "/api", HelloWeb do
+    pipe_through :api
+    resources "/users", UserController, except: [:new, :edit]
+  end
+end
+```
+
+Before running the test again, check out our new paths by running `mix
+phx.routes`.  You should see six new "/api" routes in addition to the default
+page controller route:
+
+```console
+$ mix phx.routes
+Compiling 6 files (.ex)
+page_path  GET     /               HelloWeb.PageController :index
+user_path  GET     /api/users      HelloWeb.UserController :index
+user_path  GET     /api/users/:id  HelloWeb.UserController :show
+user_path  POST    /api/users      HelloWeb.UserController :create
+user_path  PATCH   /api/users/:id  HelloWeb.UserController :update
+           PUT     /api/users/:id  HelloWeb.UserController :update
+user_path  DELETE  /api/users/:id  HelloWeb.UserController :delete
 ```
 
-We should get a new error now. Running the test informs us we don't have a `UserController`. Let's add it, along with the `index/2` action we're testing. Our test description has us returning all users:
+
+
+We should get a new error now. Running the test informs us we don't have a `HelloWeb.UserController`. Let's add it, along with the `index/2` action we're testing. Our test description has us returning all users:
 
 ```elixir
+# lib/hello_web/controllers/user_controller.ex
+
 defmodule HelloWeb.UserController do
   use HelloWeb, :controller
-
-  alias Hello.{Accounts.User, Repo}
+  alias Hello.Accounts
 
   def index(conn, _params) do
-    users = Repo.all(User)
-    render conn, "index.json", users: users
+    users = Accounts.list_users()
+    render(conn, "index.json", data: users)
```

Comment:
I don't think we should rename the `users` key to `data`. `users` is far more descriptive.(

Reply:
Agreed.  Not sure why I started doing that.  I think at some point I was working towards getting "data" in the output JSON, and for some reason it crept upstream.

Reply:
fixed